File: Backend/TrackrAPI/TrackrAPI/Models/ProgramacionExamen.cs
using System;
using System.Collections.Generic;

namespace TrackrAPI.Models
{
    public partial class ProgramacionExamen
    {
        public ProgramacionExamen()
        {
            Examen = new HashSet<Examen>();
        }

        public int IdProgramacionExamen { get; set; }
        public int IdTipoExamen { get; set; }
        public int IdUsuarioResponsable { get; set; }
        public int? IdProyectoElementoTecnica { get; set; }
        public string? Clave { get; set; }
        public double? Duracion { get; set; }
        public int? CantidadParticipantes { get; set; }
        public DateTime? FechaExamen { get; set; }
        public TimeSpan? HoraExamen { get; set; }
        public DateTime? FechaAlta { get; set; }
        public bool? Estatus { get; set; }

        public virtual ProyectoElementoTecnica? IdProyectoElementoTecnicaNavigation { get; set; }
        public virtual TipoExamen IdTipoExamenNavigation { get; set; } = null!;
        public virtual Usuario IdUsuarioResponsableNavigation { get; set; } = null!;
        public virtual ICollection<Examen> Examen { get; set; }
    }
}

File: Frontend/trackr-administrador/src/app/shared/models/examen/programacion-examen.ts
import { Time } from "@angular/common";

export class ProgramacionExamen{

    public idProgramacionExamen: number;
    public idTipoExamen: number;
    public idProyectoElementoTecnica: number;
    public tipoExamen: string;
    public idUsuarioResponsable: number;
    public usuarioResponsable: string;
    public clave: string;
    public duracion: number;
    public cantidadParticipantes: number;
    public fechaExamen: string;
    public horaExamen: string;
    public fechaAlta: Date;
    public estatus: boolean;
    public promedio: number;
    public idsPadecimiento : number[];

    public participantes: number[];

    public porcentajeAvance: number;

    constructor() {}
}
